src/components/data/UploadMenuBtn.js

```diff
@@ -5,26 +5,73 @@
  * when clicked
  */
 
-import React, { useState } from "react";
+import React, { useEffect, useState, useCallback } from "react";
 
-import { build, getMessage } from "@cyverse-de/ui-lib";
-import { Button, Hidden, makeStyles, Menu, MenuItem } from "@material-ui/core";
+import {
+    build,
+    getMessage,
+    formatMessage,
+    announce,
+    withI18N,
+} from "@cyverse-de/ui-lib";
+import {
+    Button,
+    Hidden,
+    makeStyles,
+    Menu,
+    MenuItem,
+    Typography,
+    useTheme,
+} from "@material-ui/core";
 import { Publish as UploadIcon } from "@material-ui/icons";
 
 import UploadDialog from "../uploads/dialog";
 
 import ids from "./ids";
 import styles from "./styles";
+import { useUploadTrackingState } from "../../contexts/uploadTracking";
+import { injectIntl } from "react-intl";
+import intlData from "./messages";
 
 const useStyles = makeStyles(styles);
 
 function UploadMenuBtn(props) {
-    const { baseId } = props;
+    const { baseId, intl } = props;
+    const theme = useTheme();
     const classes = useStyles();
+    const tracker = useUploadTrackingState();
 
     const [uploadAnchor, setUploadAnchor] = useState(null);
     const [uploadDialogOpen, setUploadDialogOpen] = useState(false);
 
+    const viewUploadQueue = useCallback(() => {
+        return (
+            <Button
+                variant="outlined"
+                onClick={() => setUploadDialogOpen(true)}
+            >
+                <Typography
+                    variant="button"
+                    style={{ color: theme.palette.primary.contrastText }}
+                >
+                    {formatMessage(intl, "uploadQueue")}
+                </Typography>
+            </Button>
+        );
+    }, [intl, theme.palette.primary.contrastText]);
+
+    useEffect(() => {
+        if (tracker.uploads.length > 0) {
+            announce({
+                text: `${tracker.uploads.length} ${formatMessage(
```

Comment:
I think the DE snackbar component will accept anything, not just a string.  So this could be updated to:
```
text: getMessage("filesQueuedForUploadMsg", {values: {total: tracker.uploads.length}})

// messages.js
filesQueuedForUploadMsg: "{total} file(s) queued for upload"
```

And then in that case, we can get rid of `injectIntl`

Reply:
Hmm That doesn't seems to work. I can give it another shot later.

Reply:
Even if `getMessage` won't work, `formatMessage` will take an optional `value` param.

So `filesQueuedForUploadMsg` can still be "{total} file(s) queued for upload", then this `text` can be updated to `formatMessage(intl, "filesQueuedForUploadMsg", {total: tracker.uploads.length})`.